conans/client/command.py

```diff
@@ -1000,8 +1008,29 @@ def search(self, *args):
         info = None
 
         try:
-            if ref:
+            if not v2_blocked and args.revisions:
+                try:
+                    pref = PackageReference.loads(args.pattern_or_reference)
+                except (TypeError, ConanException):
+                    pass
+                else:
+                    info = self._conan.get_package_revisions(pref.full_repr(),
+                                                             remote_name=args.remote)
+
+                if not info:
```

Comment:
I don't like that, because any exception raised in the except will say "an exception raised while other exception raised bla bla bla" and it hides a lot the origin of the issue.